app/server/api-generator/src/test/java/io/syndesis/server/api/generator/soap/parser/AbstractXmlSchemaExtractorTest.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.ws.commons.schema.XmlSchema;
+import org.apache.ws.commons.schema.XmlSchemaCollection;
+import org.apache.ws.commons.schema.XmlSchemaSerializer;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Unit tests for {@link XmlSchemaExtractor}.
+ */
+@RunWith(Parameterized.class)
+public abstract class AbstractXmlSchemaExtractorTest {
+
+    public static final String TEST_SCHEMA = "/soap/parser/test-schema.xsd";
+    protected static SchemaCollection sourceSchemas;
+
+    protected XmlSchemaExtractor xmlSchemaExtractor;
+
+    @BeforeClass
+    public static void setupClass() throws XMLStreamException, MalformedURLException, SAXException {
+        if (sourceSchemas == null) {
+            final InputSource inputSource = new InputSource(AbstractXmlSchemaExtractorTest.class
+                .getResourceAsStream(TEST_SCHEMA));
+            final XmlSchemaCollection schemaCollection = new XmlSchemaCollection();
+            schemaCollection.read(inputSource);
+            sourceSchemas = new SchemaCollection(schemaCollection);
+        }
+    }
+
+    @Before
+    public void setup() {
+        final SchemaCollection targetSchemas = new SchemaCollection();
+        xmlSchemaExtractor = new XmlSchemaExtractor(targetSchemas, sourceSchemas);
+    }
+
+    protected void validateTargetSchemas(String testNamespace) throws XmlSchemaSerializer.XmlSchemaSerializerException, IOException, SAXException {
+
+        final XmlSchema[] schemas = xmlSchemaExtractor.getTargetSchemas().getXmlSchemas();
+
+        for (XmlSchema targetSchema : schemas) {
+            final String targetNamespace = targetSchema.getTargetNamespace();
+            // skip XSD and XML namespaces, which break in StaxUtils.toString below
+            if (targetNamespace.equals(XMLConstants.XML_NS_URI) || targetNamespace.equals(XMLConstants.W3C_XML_SCHEMA_NS_URI)) {
+                continue;
+            }
+            final Document schemaDocument = targetSchema.getSchemaDocument();
+            if (targetNamespace.isEmpty()) {
+                // remove invalid xmlns:tns="" attribute or it breaks StaxUtils.toString below
+                schemaDocument.getDocumentElement().removeAttribute("xmlns:tns");
+            }
+            final String schemaString = StaxUtils.toString(schemaDocument);
+
+            // validate against schema XSD
+            XmlSchemaTestHelper.validateSchema(schemaString);
+
+            if (testNamespace.equals(targetNamespace)) {
+                // must have element in test namespace
+                assertThat(targetSchema.getElements()).isNotEmpty();
+            }
+
+            // all schemas must not be empty
+            assertThat(targetSchema.getItems()).isNotEmpty();
```

Comment:
This test doesn't test much, the values can't be tested with `isNotEmpty`, that's too loose of a check, we need to check against correct expected values.

Reply:
Added additional check based on xmlunit to make sure the extracted types match the source schema.